### PR TITLE
Ffmpeg encoder refactor and snickers server using PORT via config.json (resolves #110, #123, #124) 

### LINKED ISSUE #110: snickers server is not using config.json PORT on http server
refs https://github.com/snickers/snickers/blob/master/main.go#L19

### PR DESCRIPTION
**Pull request by dayvson**
This PR resolves #110 #123 #124

### CODE REVIEW

File: encoders/ffmpeg.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package encoders
 
 import (
 	"errors"
+	// "fmt"
```
Comment (flavioribeiro) [inline encoders/ffmpeg.go]: can you remove this line?

File: encoders/ffmpeg.go [hunk 2]
```diff
@@ -130,66 +121,114 @@ func FFMPEGEncode(logger lager.Logger, dbInstance db.Storage, jobID string) erro
...
-	for i := 0; i < outputCtx.StreamsCnt(); i++ {
-		ist, err := inputCtx.GetStream(0)
-		if err != nil {
-			return err
-		}
-		ost, err := outputCtx.GetStream(streamMap[ist.Index()])
-		if err != nil {
-			return err
-		}
+func getOutputAndInputStream(inputCtx *gmf.FmtCtx, outputCtx *gmf.FmtCtx, streamMap map[int]int, inputIndex int) (*gmf.Stream, *gmf.Stream, error) {
```
Comment (flavioribeiro) [inline encoders/ffmpeg.go]: what if we have a `getInputStream` and `getOutputStream`?

Comment (dayvson) [inline encoders/ffmpeg.go]: What about getStream and we pass the context and the index

**Comment by flavioribeiro** [comment]
LGTM, Thanks!

**Pull request merged**